File: mobile/src/main/java/rs/readahead/washington/mobile/domain/entity/uwazi/UwaziEntityInstance.java
package rs.readahead.washington.mobile.domain.entity.uwazi;

import java.io.Serializable;

public class UwaziEntityInstance implements Serializable{
    private long id = -1;
    private long updated;
    private CollectTemplate template;
    private UwaziEntityStatus status = UwaziEntityStatus.UNKNOWN;


    public long getId() {
        return id;
    }
    public void setId(long id) {
        this.id = id;
    }

    public long getUpdated() {
        return updated;
    }
    public void setUpdated(long updated) {
        this.updated = updated;
    }

    public CollectTemplate getCollectTemplate() {
        return template;
    }
    public void setCollectTemplate(CollectTemplate template) {
        this.template = template;
    }

    public UwaziEntityStatus getStatus() {
        return status;
    }
    public void setStatus(UwaziEntityStatus status) {
        this.status = status;
    }
}

File: mobile/src/main/java/rs/readahead/washington/mobile/domain/entity/uwazi/UwaziEntityInstance.kt
package rs.readahead.washington.mobile.domain.entity.uwazi

import rs.readahead.washington.mobile.domain.entity.EntityStatus
import rs.readahead.washington.mobile.domain.entity.collect.FormMediaFile
import rs.readahead.washington.mobile.domain.entity.collect.FormMediaFileStatus


data class UwaziEntityInstance(
    var id: Long = -1,
    var updated: Long = 0,
    var collectTemplate: CollectTemplate? = null,
    var relationShipEntities: List<Value> = emptyList(),
    var metadata: Map<String, List<Any>> = mutableMapOf(),
    var status : EntityStatus = EntityStatus.UNKNOWN,
    var widgetMediaFiles: List<FormMediaFile> = emptyList(),
    var relationShipEntitiesSelected: List<String> = emptyList(),
    var formPartStatus : FormMediaFileStatus = FormMediaFileStatus.UNKNOWN,
    var template: String = "",
    var title: String = "",
    var type: String = "entity") {
    var clonedId: Long = 0
}